sdk/src/wallet/wallet/mod.rs

```diff
@@ -22,47 +22,62 @@ use crate::wallet::events::{
 #[cfg(feature = "storage")]
 use crate::wallet::storage::manager::StorageManager;
 use crate::{
-    client::{secret::SecretManager, verify_mnemonic, Client},
+    client::{
+        secret::{SecretManage, SecretManager},
+        verify_mnemonic, Client,
+    },
     wallet::account::{builder::AccountBuilder, operations::syncing::SyncOptions, types::AccountBalance, Account},
 };
 
 /// The wallet, used to create and get accounts. One wallet can hold many accounts, but they should
 /// all share the same secret_manager type with the same seed/mnemonic.
-#[derive(Debug, Clone)]
-pub struct Wallet {
-    pub(crate) inner: Arc<WalletInner>,
+#[derive(Debug)]
+pub struct Wallet<S: SecretManage = SecretManager> {
```

Comment:
Do we need a type alias in lib.rs if we already have a default here?

Reply:
This, unfortunately, is only applied in impl blocks.